### PR TITLE
Fixes #8635: Having a report-separator in some NCF Methods causes the report not show up in "rudder agent run" output

### PR DESCRIPTION
**Pull request by amousset**
https://www.rudder-project.org/redmine/issues/8635

### CODE REVIEW

**Comment by amousset** [comment]
WIP

**Comment by matya** [comment]
Looks nice

**Comment by amousset** [comment]
Commit modified

**Comment by amousset** [comment]
Based on #69, with a more strict parsing (particularly if one of the first fields contains something matching [#@][#@]), and also added a warn message when lines looking like reports could not be parsed.

File: share/lib/reports.awk [hunk 1]
```diff
@@ -15,20 +18,45 @@ BEGIN {
...
+    ##
+    # r[2]  -> technique
+    # r[3]  -> result
+    # r[7]  -> component
+    # r[8]  -> key
+    # r[11] -> message
+    ##
+
+    # the rest is the message
+    message = substr($0, length($1) + 3);
```
Comment (peckpeck) [inline share/lib/reports.awk]: isn't this simply $2 ?

Comment (amousset) [inline share/lib/reports.awk]: Not if the message itself contains a "@#". The idea is to treat what comes after the first "@#" as a verbatim string.

Comment (peckpeck) [inline share/lib/reports.awk]: ok

File: share/lib/reports.awk [hunk 2]
```diff
@@ -15,20 +18,45 @@ BEGIN {
...
+    # $1 is the report, the rest is the message
+    # split the first part of the report
+    nf_report = split($1, r, "##|@@");
+
+    ##
+    # r[2]  -> technique
+    # r[3]  -> result
+    # r[7]  -> component
+    # r[8]  -> key
+    # r[11] -> message
```
Comment (peckpeck) [inline share/lib/reports.awk]: seems wrong, 'message' is the message

**Comment by amousset** [comment]
Commit modified

File: share/lib/reports.awk (hunk 2 above)
Comment (amousset) [inline share/lib/reports.awk]: indeed, fixed.

**Comment by peckpeck** [comment]
Good

**Comment by amousset** [comment]
OK, merging this PR

**Pull request merged**